src/main/java/org/zalando/nakadi/domain/BatchItem.java

```diff
@@ -58,4 +59,8 @@ public void updateStatusAndDetail(final EventPublishingStatus publishingStatus,
         response.setPublishingStatus(publishingStatus);
         response.setDetail(detail);
     }
+
+    public int getEventSize() {
+        return event.toString().getBytes(StandardCharsets.UTF_8).length;
```

Comment:
But we can try this way and see the impact. I think we can keep it as it is and test the impact on staging.